src/lib.rs

```diff
@@ -133,9 +208,23 @@ impl Config {
         if token.access_token.len() != 0 {
             Ok(token)
         } else if error.len() > 0 {
-            Err(format!("error `{}`: {}, see {}", error, error_desc, error_uri))
+            let token_error = TokenError { error, error_description, error_uri };
```

Comment:
craaaaazy :boom:

(but yeah, this means the crate will only work with the latest stable Rust - is that ok?)